src/sentry/tasks/data_export.py

```diff
@@ -24,13 +26,15 @@ def assemble_download(data_export):
         with tempfile.TemporaryFile() as tf:
             # Process the query based on its type
             if data_export.query_type == ExportQueryType.DISCOVER_V2:
-                process_discover_v2(data_export, tf)
-                return
+                file_name = process_discover_v2(data_export, tf)
             elif data_export.query_type == ExportQueryType.BILLING_REPORT:
-                process_billing_report(data_export, tf)
-                return
+                file_name = process_billing_report(data_export, tf)
             elif data_export.query_type == ExportQueryType.ISSUE_BY_TAG:
                 file_name = process_issue_by_tag(data_export, tf)
+            else:
+                raise NotImplementedError(
+                    u"Unknown query_type of {}".format(data_export.query_type)
+                )
```

Comment:
Realistically, this should never _ever_ trigger since the serializer receiving the endpoint response checks and sends a 400 if it isn't a valid query_type.